### ISSUE TITLE
`amplify codegen model` does not escape Swift keywords in generated `enum`s

### ISSUE THREAD

**Issue by cpiro** [aws-amplify/amplify-cli#3298]
In the latest version 4.13.1, if `schema.graphql` contains an `enum` value that coincides with a Swift keyword, the generated output of `amplify codegen model` is invalid. For example:

```graphql
enum PostType {
  public
  private
}
```

translates to this invalid Swift:

```swift
public enum PostType: String {
  case public
  case private
}
```

### Expected behavior
The output should be valid Swift. The older Swift codegen correctly encloses these keywords in backticks:

```swift
public enum PostType: [...] {
  case `public`
  case `private`
}
```

### Versions
```
$ uname
Darwin
$ amplify -v
4.13.1
$ node -v
v13.6.0
$ swift --version
Apple Swift version 5.1.3 (swiftlang-1100.0.282.1 clang-1100.0.33.15)
Target: x86_64-apple-darwin19.2.0
```

### To Reproduce
Install the CLI, then run `amplify init`, `amplify add api`, and `amplify codegen model`. I choose ios as the "type of app that you're building".

```
$ npm install -g @aws-amplify/cli@latest
[...]
+ @aws-amplify/cli@4.13.1

$ amplify init
[...]
? Choose the type of app that you're building ios
[...]

$ cat > minimal.graphql
enum PostType {
  public
  private
}

$ amplify add api
[...]
? Provide your schema file path: minimal.graphql
[...]

$ amplify codegen model

GraphQL schema compiled successfully.

Edit your schema at [...] or place .graphql files in a directory at [...]
Successfully generated models. Generated models can be found [...]

$ cat amplify/generated/models/PostType.swift
// swiftlint:disable all
import Amplify
import Foundation

public enum PostType: String {
  case public
  case private
}

**Comment by yuth** [comment]
We have released the fix for this in Amplify CLI v 4.17.2.

**Comment by vg-identance** [comment]
Same issue with `init` keyword
